ftw/testbrowser/core.py

```diff
@@ -803,6 +804,44 @@ def get_mechbrowser(self):
                     'X-zope-handle-errors', 'False'))
         return self.mechbrowser
 
+    def parse_as_html(self, html=None):
+        """Parse the response document with the HTML parser.
+
+        .. seealso:: :py:mod:`ftw.testbrowser.core.Browser.parse_as_xml`
+        .. seealso:: :py:mod:`ftw.testbrowser.core.Browser.parse`
+
+        :param html: The HTML to parse (default: current response).
+        :type html: string
+        """
+        return self._load_html(html or self.response, lxml.html.parse)
+
+    def parse_as_xml(self, xml=None):
+        """Parse the response document with the XML parser.
+
+        .. seealso:: :py:mod:`ftw.testbrowser.core.Browser.parse_as_html`
+        .. seealso:: :py:mod:`ftw.testbrowser.core.Browser.parse`
+
+        :param xml: The XML to parse (default: current response).
+        :type xml: string
+        """
+        return self._load_html(xml or self.response, lxml.etree.parse)
+
+    def parse(self, xml_or_html):
+        """Parse XML or HTML with the default parser.
+        For XML mime types the XML parser is used, otherwise the HTML parser.
+
+        .. seealso:: :py:mod:`ftw.testbrowser.core.Browser.parse_as_html`
+        .. seealso:: :py:mod:`ftw.testbrowser.core.Browser.parse_as_xml`
+
+        :param xml: The XML or HTML to parse.
+        :type xml: string
+        """
+
+        if self.mimetype in ('text/xml', 'application/xml'):
```

Comment:
This could have bigger impact, since this is also kind of HTML.
As far as I know the ``lxml.html`` have custom ``Element`` implementations for certain HTML tags, which would then not be used for xhtml, right?